File: server/src/MerchWebsite.API/Models/DTOs/OrderItemDto.cs
// ~/Projects/MerchWebsite/server/src/MerchWebsite.API/Models/DTOs/OrderItemDto.cs
namespace MerchWebsite.API.Models.DTOs
{
    public class OrderItemDto
    {
        public int ProductId { get; set; }
        public string ProductName { get; set; } = string.Empty; // Name when ordered
        public string? ProductImageUrl { get; set; } // ImageUrl when ordered
        public decimal Price { get; set; } // Price per unit when ordered
        public int Quantity { get; set; }
        public decimal TotalPrice => Price * Quantity;
    }
}

File: client/src/app/models/order/order-item.dto.ts
// client/src/app/models/order/order-item.dto.ts
// Matches backend OrderItemDto
export interface OrderItemDto {
  productId: number;
  productName: string;
  productImageUrl: string | null; // Allow null
  price: number;
  quantity: number;
  totalPrice: number; // Calculated property
}
